### ISSUE TITLE
Webmentions not displayed when Semantic Linkbacks is activated

### ISSUE THREAD

**Issue by piperhaywood** [pfefferle/wordpress-webmention#318]
It’s very strange, for a long time Webmentions were working great on my site (great plugin!), as was [Semantic Linkbacks](https://github.com/pfefferle/wordpress-semantic-linkbacks) (also a great plugin!). But they stopped displaying suddenly at some point. Unfortunately I’m not sure exactly when, though I think it was at some point in Feb 2021 or before.

While debugging the issue, I tried deactivating Semantic Linkbacks and this fixed the problem. I could just keep that plugin deactivated, but I’d love to keep using Semantic Linkbacks if possible.

This is the code for my `comments.php`: https://github.com/piperhaywood/commonplace-wp-theme/blob/master/commonplace-theme/comments.php

And these are some examples of where the webmentions should be displayed but aren’t:

Post: https://piperhaywood.com/four-plus-month-update/
Webmention: https://jeremyfelt.com/notes/1d2a1639287888/

Post: https://piperhaywood.com/rietveld-brigham-didion/
Webmention: https://nicksimson.com/posts/2021-commonplace-books/

Post: https://piperhaywood.com/failures-low-moments/
Webmention: http://cdevroe.com/2021/06/17/wis-81/

Post: https://piperhaywood.com/on-personal-sites-adios-analytics/
Webmention: https://doubleloop.net/2020/09/06/a-home-on-the-web/

Post: https://piperhaywood.com/feet-on-the-ground-head-in-the-sky/
Webmention: https://doubleloop.net/2020/03/28/6697/

Post: https://piperhaywood.com/switching-from-google-analytics-to-matomo-f-k-a-piwik-on-wordpress/
Webmention: https://www.jvt.me/mf2/2020/02/mrnf9/
Webmention: https://doubleloop.net/2020/01/10/6510/

Post: https://piperhaywood.com/wake-up-and-lie-there-and-think/
Webmention: https://doubleloop.net/2019/11/11/6369/

Note that I still seem to be receiving the webmentions successfully, they're stored correctly in the database. They just aren’t displayed if Semantic Linkbacks is active. Not sure where to go from here with debugging!

**Comment by pfefferle** [comment]
Hey @piperhaywood 

it seems to work on this site: https://piperhaywood.com/switching-from-google-analytics-to-matomo-f-k-a-piwik-on-wordpress/ , so I think it might be a problem with "Discussion is closed". Can you try to enable comments here: https://piperhaywood.com/wake-up-and-lie-there-and-think/ to verify?

**Comment by piperhaywood** [comment]
Hey @pfefferle thanks for taking a look at this!

I enabled comments on https://piperhaywood.com/wake-up-and-lie-there-and-think/, and it’s strange, the webmention that is related to that post almost seems to be displayed by the WordPress `comment_form()` function ([line 54](https://github.com/piperhaywood/commonplace-wp-theme/blob/cc1c97e2365b3cd4a9a5c1f0be4338acc3dad363/commonplace-theme/comments.php#L54) in my `comments.php` file; quick note about that, lines 49-50 are unnecessary and I need to rip them out!) based on where it is located, within the “Leave a reply” section. On top of that, it’s missing a lot of the contents. Screenshot below:

<img width="871" alt="Screen Shot 2022-01-17 at 13 27 37" src="https://user-images.githubusercontent.com/4711611/149822159-5db8e641-5cef-4095-a6fd-f1d26236222b.png">

Then when I temporarily disable Semantic-Linkbacks, the webmention is in the right place and has all of the relevant contents. Screenshot:

<img width="871" alt="Screen Shot 2022-01-17 at 13 27 54" src="https://user-images.githubusercontent.com/4711611/149822184-4af55db8-c4a0-4c9b-9812-13f0dbb66ca9.png">

For now, I’ve left Semantic-Linkbacks activated, and comments are enabled on the “5:30am — wake up and lie there and think” post. But let me know if any other info would be useful or if you think trying anything else would be worthwhile!

**Comment by pfefferle** [comment]
Of you want them to be displayed as full comments, you have to go to the settings and disable facepiles for all comment types. I will have a look, why it does not show when comments are disabled.

**Comment by piperhaywood** [comment]
Ah interesting... When I disabled facepiles, the comment was displayed (as you described) but also it is still present even when I close comments (which is what I was hoping for!). Maybe I enabled facepiles at some point a while ago without realizing the knock-on effect, that the comment wouldn’t be displayed.

So to be clear, I think everything is working A-OK in your plugins, the issue is that I wasn’t expecting the comment to come through in a facepile since I had forgotten that I had it enabled. Since the facepiles seem to be linked to the `comment_form()` function and I am only displaying the comment form when comments are open, the facepile wasn’t showing when comments were disabled.

I’ll go ahead and close this. Thank you so much for helping me get to the bottom of it!
